annnufan/assignment3/test_words_finder.cpp

```diff
@@ -2,32 +2,35 @@
 #include "codeu_test_lib.h"
 #include "words_finder.cpp"
 
-std::vector<std::vector<char>> ex_grid({{'a', 'a', 'r'}, {'t', 'c', 'd'}});
-std::vector<std::vector<char>> empty_grid;
-std::vector<std::vector<char>> grid_1({{'a', 'o', 'b', 'e', 'r'}, {'c', 'a', 'd', 'e', 't'}, {'b', 'i', 'v', 'i', 'n'}});
-
-dictionary ex_dict({"car", "card", "cart", "cat"});
-dictionary empty_dict;
-dictionary dict_1({"tree", "cart", "bee", "bed", "cider", "beer", "teer", "dinner", "baobab", "vine", "tear", "horse", "bob"});
-
-std::set<std::string> correct_ans({"car", "card", "cat"});
-std::set<std::string> empty_ans;
-std::set<std::string> ans_1({"tree", "bee", "bed", "cider", "beer", "teer", "vine"});
-
-
 void test_from_example() {
-	EXPECT_EQ(find_words_from_dict_on_grid(ex_dict, ex_grid), correct_ans);
+	dictionary ex_dict({"car", "card", "cart", "cat"});
+	std::vector<std::vector<char>> ex_grid({{'a', 'a', 'r'}, {'t', 'c', 'd'}});
+	std::set<std::string> ex_ans({"car", "card", "cat"});
+
+	EXPECT_EQ(find_words_from_dict_on_grid(ex_dict, ex_grid), ex_ans);
 }
 
 void test_empty_dict() {
+	dictionary empty_dict;
+	std::vector<std::vector<char>> ex_grid({{'a', 'a', 'r'}, {'t', 'c', 'd'}});
+	std::set<std::string> empty_ans;
+
 	EXPECT_EQ(find_words_from_dict_on_grid(empty_dict, ex_grid), empty_ans);
 }
 
 void test_empty_grid() {
+	dictionary ex_dict({"car", "card", "cart", "cat"});
+	std::vector<std::vector<char>> empty_grid;
+	std::set<std::string> empty_ans;
+
 	EXPECT_EQ(find_words_from_dict_on_grid(ex_dict, empty_grid), empty_ans);
 }
 
 void test_double_usage_cell() {
+	dictionary dict_1({"tree", "cart", "bee", "bed", "cider", "beer", "teer", "dinner", "baobab", "vine", "tear", "horse", "bob"});
+	std::vector<std::vector<char>> grid_1({{'a', 'o', 'b', 'e', 'r'}, {'c', 'a', 'd', 'e', 't'}, {'b', 'i', 'v', 'i', 'n'}});
+	std::set<std::string> ans_1({"tree", "bee", "bed", "cider", "beer", "teer", "vine"});
+

```

Comment:
I think that this way, each test is much more readable, as you can easily see its input and expected output.
Well done! :+1: 

-----
FYI
In other cases, where the input initialization is very long and tests end up having a lot of common "copy-pasted" code, it is possible to extract the common part to a function. That's not the case here, because code defining the input is quite short (2 lines), and having the code a bit copied is not too bad.
For example, lines 6 and 22 are the same, but still I prefer it this way, not extracted to some common function or variable, so each test can be independently read. Do you agree?

Reply:
Yes, I agree, that this type of declaration in this cases easier to understand